sdk/rust/src/hbar.rs

```diff
@@ -0,0 +1,195 @@
+use std::fmt::{Display, Formatter};
+use std::str::FromStr;
+
+use derive_more::{
+    Add,
+    AddAssign,
+    Div,
+    DivAssign,
+    Mul,
+    MulAssign,
+    Neg,
+    Sub,
+    SubAssign,
+};
+use rust_decimal::prelude::*;
+use serde_with::{
+    DeserializeFromStr,
+    SerializeDisplay,
+};
+
+use crate::Error;
+
+// TODO: add tests
+
+#[derive(SerializeDisplay, Copy, DeserializeFromStr, Hash, PartialEq, Eq, Clone, Add, Sub, Mul, Div, Neg, AddAssign, SubAssign, MulAssign, DivAssign)]
+pub struct Hbar(i64);
+
+pub type Tinybar = i64;
+
+#[repr(i64)]
+#[derive(Debug, SerializeDisplay, Copy, DeserializeFromStr, Hash, PartialEq, Eq, Clone)]
+pub enum HbarUnit {
+    /**
+     * The atomic (smallest) unit of hbar, used natively by the Hedera network.
+     * <p>
+     * It is equivalent to <sup>1</sup>&frasl;<sub>100,000,000</sub> hbar.
+     */
+    Tinybar = 1,
+
+    /**
+     * Equivalent to 100 tinybar or <sup>1</sup>&frasl;<sub>1,000,000</sub> hbar.
+     */
+    Microbar = 100,
+
+    /**
+     * Equivalent to 100,000 tinybar or <sup>1</sup>&frasl;<sub>1,000</sub> hbar.
+     */
+    Millibar = 100_000,
+
+    /**
+     * The base unit of hbar, equivalent to 100 million tinybar.
+     */
+    Hbar = 100_000_000,
+
+    /**
+     * Equivalent to 1 thousand hbar or 100 billion tinybar.
+     */
+    Kilobar = 1_000 * 100_000_000,
+
+    /**
+     * Equivalent to 1 million hbar or 100 trillion tinybar.
+     */
+    Megabar = 1_000_000 * 100_000_000,
+
+    /**
+     * Equivalent to 1 billion hbar or 100 quadillion tinybar.
+     * <p>
+     * The maximum hbar amount supported by Hedera in any context is ~92 gigabar
+     * (2<sup>63</sup> tinybar); use this unit sparingly.
+     */
+    Gigabar = 1_000_000_000 * 100_000_000,
+}
+
+impl HbarUnit {
+    pub fn tinybars(self) -> Tinybar {
+        self as Tinybar
+    }
+
+    pub fn symbol(self) -> &'static str {
+        match self {
+            HbarUnit::Tinybar => "tℏ",
+            HbarUnit::Microbar => "μℏ",
+            HbarUnit::Millibar => "mℏ",
+            HbarUnit::Hbar => "ℏ",
+            HbarUnit::Kilobar => "kℏ",
+            HbarUnit::Megabar => "Mℏ",
+            HbarUnit::Gigabar => "Gℏ",
+        }
+    }
+}
+
+impl Display for HbarUnit {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.symbol())
```

Comment:
`Formatter::write_str` is decently more performant and slightly smaller than `write!()` because of formatting machinery
```suggestion
        f.write_str(self.symbol);
```